tutorials/haskell/golden-testing-in-haskell-with-hspec-gold/tutorial.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Golden testing in Haskell with Hspec-Golden
+published: 2016-08-10
+tags: haskell, golden testing, snapshot testing, expect testing
+libraries: hspec-golden-1.0.0
+language: haskell
+author: Adrian Vera
+author-name: Adrian Vera
+twitter-profile: adrian_med
+github-profile: medAdrian
+description: A golden test is an IO action that writes its result to a file. To pass the test, this output file should be identical to the corresponding golden file, which contains the correct result for the test.
+---
+
+Golden tests are like unit tests, with the difference that the expected output is stored in a separate file. 
+
+Golden tests can also be known as:
+* Golden master testing
+* Approval testing
+* Snapshot testing
+* Expect testing
+
+There are many forms of getting the same result with different approaches, as the [Jest](https://jestjs.io/docs/en/snapshot-testing.html) snapshot testing,
+comparing two screenshots as [Wraith](https://github.com/BBC-News/wraith) or with the [Approval testing method](https://dzone.com/articles/approval-tests-alternative).
+
+##The Main idea
+A golden test is just an **IO action** that writes its result to a file.
+To pass the test, the output **file** should be identical to the corresponding golden file,
+that contains the correct result for the given test.
+
+####‘A Golden File is the absolute source of truth. You validate your response against the Golden File. In a nutshell, the Golden file contains the response you expect from your test’
+
+##Changes on test flow
+The workflow for running the tests now becomes the following:
+
+* Run a test and make sure it fails.
```

Comment:
@MedAdrian Not sure if this first step is correct. As you might have seen, a golden always passes the first time, because it copies the actual output into the golden file.

Reply:
You're right it would better be something like:
`Write some execution for the part of the subject under test, this can be a small unit test, the goal here is to click a button to run your code.`

Does it make sense?,
also, are the next steps accurate enough?

* Review the output from your test to see if it is accurate.
* Update the existing golden file (or create new ones).
* Re-run your tests to ensure that the test does not fail.